### PR TITLE
Align persistent cache blocks with device pages

### PR DESCRIPTION
**Pull request by assafein**
When persistent cache appends blocks to a file, these blocks may spread over several device pages. In order to read these blocks, the cache has to read their corresponding pages entirely, potentially creating significant read amplification. For example, to read a 4K block that does not start at the beginning of a page (and thus, spreads over two pages), the cache will actually have to read 8K.
To reduce read bandwidth from the persistent cache device, this diff introduces an option to align blocks with pages, such that each block will spread over the minimal number of pages possible. When needed, a page will be padded with zeros in order to align a block with the beginning of next page.

### CODE REVIEW

File: utilities/persistent_cache/block_cache_tier_file.cc [hunk 1]
```diff
@@ -316,6 +318,27 @@ bool WriteableCacheFile::Append(const Slice& key, const Slice& val, LBA* lba) {
...
 
+  if (page_alignment_ > 0) {
+    assert(page_alignment_ % 2 == 0);
+
+    uint32_t min_extra_pages = (rec_size-1) / page_alignment_;
+    uint32_t start_page = disk_woff_ / page_alignment_;
+    uint32_t end_page = (disk_woff_ + rec_size) / page_alignment_;
+
+    if (end_page - start_page != min_extra_pages) {
+      assert(end_page - start_page > min_extra_pages);
```
Comment (IslamAbdelRahman) [inline utilities/persistent_cache/block_cache_tier_file.cc]: the difference is supposed to be always one, right ?

**Comment by IslamAbdelRahman** [comment]
can you please fix window build ?
https://ci.appveyor.com/project/Facebook/rocksdb/build/1.0.5916

**Comment by maysamyabandeh** [comment]
@siying Not sure why it was accepted but not landed. Since it was from long time ago, it might be obsolete already. Do you might have more context on this patch?

**Comment by siying** [comment]
@maysamyabandeh the plan was to deprecate PersistentCache so I guess we can close it. I assume that @assafein already used the data in the MyNVM paper so it is not needed anymore.

**Pull request closed without merging**